src/Drivers/Paypal/Paypal.php

```diff
@@ -100,32 +108,54 @@ public function pay() : RedirectionForm
      *
      *
      * @throws InvalidPaymentException
-     * @throws \SoapFault
+     * @throws PurchaseFailedException
      */
     public function verify() : ReceiptInterface
     {
-        $authority = $this->invoice->getTransactionId() ?? Request::input('Authority');
-        $status = Request::input('Status');
+        $transactionId = Request::input('token');
 
-        $data = [
-            'MerchantID' => $this->settings->merchantId,
-            'Authority' => $authority,
-            'Amount' => $this->invoice->getAmount(),
-        ];
+        $accessToken = $this->getAccessToken();
+        $verificationUrl = str_replace('{order_id}', $transactionId, $this->getVerificationUrl());
 
-        if ($status != 'OK') {
-            throw new InvalidPaymentException('عملیات پرداخت توسط کاربر لغو شد.');
-        }
+        $response = $this
+            ->client
+            ->request(
+                'POST',
+                $verificationUrl,
+                [
+                    "headers" => [
+                        'Content-Type' => 'application/json',
+                        'Authorization' => "Bearer $accessToken",
+                    ],
+                    "http_errors" => false,
+                ]
+            );
+        $result = json_decode($response->getBody()->getContents(), true);
 
-        $client = new \SoapClient($this->getVerificationUrl(), ['encoding' => 'UTF-8']);
-        $result = $client->PaymentVerification($data);
 
-        if ($result->Status != 100) {
-            $message = $this->translateStatus($result->Status);
-            throw new InvalidPaymentException($message, (int)$result->Status);
+        // handle possible errors
+        if ( !in_array($response->getStatusCode(), [200, 201]) )
+        {
+            if (isset($result['name']) && isset($result['message']))
+                $message = $result['name'] . ': ' . $result['message'];
+            else
+                $message = "Unknown error";
+
+            throw new PurchaseFailedException($message);
         }
 
-        return $this->createReceipt($result->RefID);
+        if (isset($result['status']) && $result['status'] != 'COMPLETED')
+            throw new PurchaseFailedException("Purchase not completed");
+
+
+        // finalize verification
+        $receipt =  $this->createReceipt($result['id']);
+
+        $receipt->detail([
```

Comment:
please review the code one more time + make sure it works properly.

```suggestion
        $receipt->detail([
```

see: https://github.com/shetabit/multipay/blob/master/src/Traits/HasDetail.php#L22

Reply:
Tested it again in a seperate project. Everything seems to be working as intended. I'm not entirely sure what was meant in the "Suggested change" part, but I added some extra details to the receipt object in the new commit.